### PR TITLE
feat: [CDS-76160]: Fix null pointers when expression is unresolved fo…

### PR DESCRIPTION
**Pull request by abhinavmittal-harness**
…r booleans

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51932)

### CODE REVIEW

**Comment by vivekkarnatiharness** [comment]
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/approval/step/harness/entities/HarnessApprovalInstance.java` line 156 at r1](https://reviewable.io/reviews/harness/harness-core/51932#-NcXE1o_5gXX6p9KLizw:-NcXE1o_5gXX6p9KLizx:b-1smgzk) ([raw file](https://github.com/harness/harness-core/blob/236cf8eefdc70153440737c2e1d54ce2b6976bcb/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/approval/step/harness/entities/HarnessApprovalInstance.java#L156)):*
> ```Java
>         HarnessApprovalInstance.builder()
>             .approvalMessage((String) specParameters.getApprovalMessage().fetchFinalValue())
>             .includePipelineExecutionHistory((boolean) ParameterFieldHelper.getBooleanParameterFieldValue(
> ```

Is this a good way to cast Boolean to boolean, generally we do like this cast right  "Boolean.TRUE.equals(value)"

**Review by abhinavmittal-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51932)* status: 3 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/approval/step/harness/entities/HarnessApprovalInstance.java` line 156 at r1](https://reviewable.io/reviews/harness/harness-core/51932#-NcXE1o_5gXX6p9KLizw:-NcXTbtFEU4bHcpjAm3u:b-po6wak) ([raw file](https://github.com/harness/harness-core/blob/236cf8eefdc70153440737c2e1d54ce2b6976bcb/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/approval/step/harness/entities/HarnessApprovalInstance.java#L156)):*
<details><summary><i>Previously, vivekkarnatiharness wrote…</i></summary><blockquote>

Is this a good way to cast Boolean to boolean, generally we do like this cast right  "Boolean.TRUE.equals(value)"
</blockquote></details>

Since `ParameterFieldHelper.getBooleanParameterFieldValue()` returns a `Boolean`, then we are fine since Java's auto-unboxing will handle converting the `Boolean` object to a primitive `boolean` value.

And the method is taking care of null case so that eliminates the issue of null pointer.

The reason for comparator you do it that way is because auto-unboxing can lead to unexpected results. For example, new Integer(5) == new Integer(5) will return false as it compares object references, not values.

**Review by puthrayaharness — APPROVED** [review]

**Comment by abhinavmittal-harness** [comment]
trigger uts

File: pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/approval/step/harness/beans/ApproversDTO.java [hunk 1]
```diff
@@ -76,7 +82,8 @@ public static ApproversDTO fromApprovers(Approvers approvers) {
     return ApproversDTO.builder()
         .userGroups(userGroups)
         .minimumCount(minimumCount)
-        .disallowPipelineExecutor((Boolean) approvers.getDisallowPipelineExecutor().fetchFinalValue())
+        .disallowPipelineExecutor(
+            (Boolean) ParameterFieldHelper.getBooleanParameterFieldValue(approvers.getDisallowPipelineExecutor()))
```
Comment (naman-goenka) [inline pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/approval/step/harness/beans/ApproversDTO.java:86]: implicit cast

**Comment by abhinavmittal-harness** [comment]
trigger review

**Comment by abhinavmittal-harness** [comment]
trigger messagecheck

**Pull request merged**